src/Config/Parser/AnnotationParser.php

```diff
@@ -257,57 +237,52 @@ private static function classAnnotationsToGQLConfiguration(
     /**
      * @throws ReflectionException
      */
-    private static function extractClassAnnotations(string $className): array
+    private static function getGraphClass(string $className): GraphClass
     {
-        if (!isset(self::$classAnnotationsCache[$className])) {
-            $annotationReader = self::getAnnotationReader();
-            $reflectionEntity = new ReflectionClass($className); // @phpstan-ignore-line
-            $classAnnotations = $annotationReader->getClassAnnotations($reflectionEntity);
-
-            $properties = [];
-            $reflectionClass = new ReflectionClass($className); // @phpstan-ignore-line
-            do {
-                foreach ($reflectionClass->getProperties() as $property) {
-                    if (isset($properties[$property->getName()])) {
-                        continue;
-                    }
-                    $properties[$property->getName()] = ['property' => $property, 'annotations' => $annotationReader->getPropertyAnnotations($property)];
-                }
-            } while ($reflectionClass = $reflectionClass->getParentClass());
-
-            $methods = [];
-            foreach ($reflectionEntity->getMethods() as $method) {
-                $methods[$method->getName()] = ['method' => $method, 'annotations' => $annotationReader->getMethodAnnotations($method)];
-            }
-
-            self::$classAnnotationsCache[$className] = [$reflectionEntity, $classAnnotations, $properties, $methods];
+        if (!isset(self::$graphClassCache[$className])) {
+            self::$graphClassCache[$className] = new GraphClass($className);
         }
```

Comment:
This can be shortened:
```suggestion
        self::$graphClassCache[$className] ??= new GraphClass($className);
```